core/src/main/scala/pureconfig/FluentConfigCursor.scala

```diff
@@ -0,0 +1,127 @@
+
+  /**
+   * Casts this cursor to an int.
+   *
+   * @return a `Right` with the int value pointed to by this cursor if the cast can be done, `Left` with a list of
+   *         failures otherwise.
+   */
+  def asInt: ConfigReader.Result[Int] = cursor.right.flatMap(_.asInt)
+
+  /**
+   * Casts this cursor to a short.
+   *
+   * @return a `Right` with the short value pointed to by this cursor if the cast can be done, `Left` with a list of
+   *         failures otherwise.
+   */
+  def asShort: ConfigReader.Result[Short] = cursor.right.flatMap(_.asShort)
+
+  /**
+   * Casts this cursor to a double.
+   *
+   * @return a `Right` with the double value pointed to by this cursor if the cast can be done, `Left` with a list of
+   *         failures otherwise.
+   */
+  def asDouble: ConfigReader.Result[Double] = cursor.right.flatMap(_.asDouble)
+
+  /**
+   * Casts this cursor to a float.
+   *
+   * @return a `Right` with the float value pointed to by this cursor if the cast can be done, `Left` with a list of
+   *         failures otherwise.
+   */
+  def asFloat: ConfigReader.Result[Float] = cursor.right.flatMap(_.asFloat)
+
+  /**
+   * Casts this cursor to a `ConfigListCursor`.
+   *
+   * @return a `Right` with this cursor as a list cursor if the cast can be done, `Left` with a list of failures
+   *         otherwise.
+   */
+  def asListCursor: ConfigReader.Result[ConfigListCursor] = cursor.right.flatMap(_.asListCursor)
+
+  /**
+   * Casts this cursor to a `ConfigObjectCursor`.
+   *
+   * @return a `Right` with this cursor as an object cursor if it points to an object, `Left` with a list of failures
+   *         otherwise.
+   */
+  def asObjectCursor: ConfigReader.Result[ConfigObjectCursor] = cursor.right.flatMap(_.asObjectCursor)
+
+  /**
+   * Returns a cursor to the config at a given path.
+   *
+   * @param segments the path of the config for which a cursor should be returned
+   * @return a `FluentConfigCursor` pointing to the provided path.
+   */
+  def at(segments: PathSegment*): FluentConfigCursor = FluentConfigCursor {
+    segments.foldLeft(this.cursor) {
+      case (Right(cur), PathSegment.Key(k)) => cur.asObjectCursor.right.flatMap(_.atKey(k))
+      case (Right(cur), PathSegment.Index(i)) => cur.asListCursor.right.flatMap(_.atIndex(i))
+      case (Left(err), _) => Left(err)
+    }
+  }
+
+  /**
+   * Casts this cursor to a `ConfigListCursor` and maps each element to a result. This method tries to map all
+   * elements, combining failures from all of them if more than one exists.
+   *
+   * @param f the function used to map elements
+   * @tparam A the result type of the elements
+   * @return a `Right` with the list obtained by mapping all elements of the list pointed to by this cursor if all
+   *         casts and mappings can be done, `Left` with a list of failures otherwise.
+   */
+  def mapList[A](f: ConfigCursor => ConfigReader.Result[A]) =
```

Comment:
Just for consistency with the other public methods in this class:

```suggestion
  def mapList[A](f: ConfigCursor => ConfigReader.Result[A]): ConfigReader.Result[List[A]] =
```